dev-libs/libmelf/libmelf-0.4.0-r2.ebuild

```diff
@@ -0,0 +1,53 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit flag-o-matic toolchain-funcs
+
+DESCRIPTION="libmelf is a library interface for manipulating ELF object files"
+HOMEPAGE="http://www.hick.org/code/skape/libmelf/"
+SRC_URI="http://www.hick.org/code/skape/${PN}/${P}.tar.gz"
+
+LICENSE="Artistic"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="static-libs"
+
+PATCHES=(
+	# This patch was gained from the elfsign-0.2.2 release
+	"${FILESDIR}"/${PN}-0.4.1-unfinal-release.patch
+	# Cleanup stuff
+	"${FILESDIR}"/${PN}-0.4.0-r1-gcc-makefile-cleanup.patch
+)
+
+src_prepare() {
+	default
+
+	# * QA Notice: The following shared libraries lack a SONAME
+	# * /usr/lib64/libmelf.so
+	sed "s/\(-shared -fPIC\)/\1 ${LDFLAGS} -Wl,-soname,libmelf.so/" -i configure || die
```

Comment:
Adding ${LDFLAGS} here was the quickest fix I could think of - but there are also patches that rework the configure script, so it might make sense to rework those to include LDFLAGS?